dingo/core/result.py

```diff
@@ -547,20 +556,22 @@ def merge(cls, parts):
         Combined Result.
         """
         dataset_dict = parts[0].to_dictionary()
-        del dataset_dict["log_evidence"]
+        if "log_evidence" in dataset_dict:
+            del dataset_dict["log_evidence"]
         samples_parts = [dataset_dict.pop("samples")]
 
         for part in parts[1:]:
             part_dict = part.to_dictionary()
-            del part_dict["log_evidence"]
+            if "log_evidence" in dataset_dict:
+                del part_dict["log_evidence"]
             samples_parts.append(part_dict.pop("samples"))
 
             # Make sure we are not merging incompatible results. We deleted the
             # log_evidence since this can differ among the sub-results. Note that this
             # will also raise an error if files were created with different versions of
             # dingo.
-            if not recursive_check_dicts_are_equal(part_dict, dataset_dict):
-                raise ValueError("Results to be merged must have same metadata.")
+            # if not recursive_check_dicts_are_equal(part_dict, dataset_dict):
+                # raise ValueError("Results to be merged must have same metadata.")
```

Comment:
This was to make sure we are not merging incompatible results. Why was it necessary to comment this?

Reply:
This along with other changes to result.py I think don't belong in this PR. I've reset the file to how it is on main.